backend/src/main/java/de/bund/digitalservice/ris/norms/application/service/VerkuendungsImportService.java

```diff
@@ -47,4 +88,200 @@ public UUID storeNormendokumentationspaket(Query query) throws IOException {
   }
+
+  @Override
+  public void processNormendokumentationspaket(
+    ProcessNormendokumentationspaketUseCase.Query query
+  ) {
+    // TODO: (Malte Laukötter, 2025-04-08) call parseAndValidate with the zip file resource
+
+    throw new RuntimeException("Not implemented yet");
+  }
+
+  // TODO: (Malte Laukötter, 2025-04-08) make private once integrated in processNormendokumentationspaket
+  public Norm parseAndValidate(Resource zipFile)
+    throws IOException, NormendokumentationspaketImportFailedException {
+    validateFileIsZipArchive(zipFile);
+    log.debug("File is zip archive.");
+
+    Map<String, byte[]> files;
+    try {
+      files = ZipUtils.unzipFileWithoutDirectories(zipFile.getInputStream());
+    } catch (IllegalArgumentException e) {
+      throw new InvalidStructureInZipFileException(e.getMessage());
+    }
+
+    Set<Dokument> dokumente = findParseAndValidateDokumente(files);
+
+    var manifestationEli = dokumente
+      .stream()
+      .findFirst()
+      .orElseThrow()
+      .getManifestationEli()
+      .asNormEli();
+
+    var binaryFiles = files
+      .entrySet()
+      .stream()
+      .filter(entry ->
+        dokumente
+          .stream()
+          .noneMatch(dokument -> dokument.getManifestationEli().getFileName().equals(entry.getKey())
+          )
+      )
+      .map(entry -> {
+        log.debug("Found binary file: {}", entry.getKey());
+
+        var mimeType = MediaTypeFactory.getMediaType(entry.getKey()).orElseThrow();
+
+        if (SUPPORTED_MIME_TYPES.stream().noneMatch(mimeType::isCompatibleWith)) {
+          throw new UnsupportedFileTypeException(entry.getKey());
+        }
+
+        return new BinaryFile(
+          DokumentManifestationEli.fromNormEli(
+            manifestationEli,
+            entry.getKey().split("\\.")[0],
+            entry.getKey().split("\\.")[1]
+          ),
+          entry.getValue()
+        );
+      })
+      .collect(Collectors.toSet());
+
+    var norm = new Norm(NormPublishState.UNPUBLISHED, dokumente, binaryFiles);
+
+    ldmlDeValidator.validateSchematron(norm);
+
+    if (
+      !norm
+        .getRechtsetzungsdokument()
+        .orElseThrow(MissingRechtsetzungsdokumentException::new)
+        .isVerkuendungsfassung()
+    ) {
+      throw new RechtsetzungsdokumentNotAVerkuendungsfassungException();
+    }
+
+    if (norm.getRegelungstexte().isEmpty() && norm.getBekanntmachungen().isEmpty()) {
+      throw new NoRegelungstextOrBekanntmachungstextException();
+    }
+
+    if (loadNormPort.loadNorm(new LoadNormPort.Command(norm.getWorkEli())).isPresent()) {
+      throw new NormExistsAlreadyException(norm.getWorkEli().toString());
+    }
+
+    log.info(
+      "Verified new norm from import: {} with {} Dokumenten and {} Binary files",
+      norm.getManifestationEli(),
+      norm.getDokumente().size(),
+      norm.getBinaryFiles().size()
+    );
+
+    return norm;
+  }
+
+  private void validateFileIsZipArchive(Resource file) {
```

Comment:
This is only relying on the MIME type detection which is inferring the type from the file extension. Though I think the Spring `MediaTypeFactory` is also sniffing the content based on file byte patterns, it is not 100% foolproof. Should we also add some magic number validation, meaning reading some first bytes?

Reply:
I know that parsing the ZIP would also tell us if it is truly a ZIP or not, but you are throwing here the `NotAZipFileException` and this one should also be thrown if when trying to parse the ZIP it comes out that it is not a ZIP right? Because right now if something goes wrong when trying to parse the ZIP we would get a `InvalidStructureInZipFileException`